src/CarbonAware.WebApi/src/Controllers/CarbonAwareController.cs

```diff
@@ -115,24 +115,47 @@ public async Task<IActionResult> GetEmissionsDataForLocationByTime([FromQuery, B
-    /// <returns>HTTP response containing the results of the data source current forecast call</returns>
+    /// <param name="location"> String array of named locations.</param>
+    /// <param name="startTime">
+    ///   Start time boundary of forecasted data points. Ignores current forecast data points before this time.
+    ///   Defaults to the earliest time in the forecast data.
+    /// </param>
+    /// <param name="endTime">
+    ///   End time boundary of forecasted data points. Ignores current forecast data points after this time.
+    ///   Defaults to the latest time in the forecast data.
+    /// </param>
+    /// <param name="windowSize">
+    ///   The estimated duration (in minutes) of the workload.
+    ///   Defaults to the duration of a single forecast data point.
+    /// </param>
+    /// <remarks>
+    ///   This endpoint fetches the most recent forecast for all provided locations and calculates the optimal 
+    ///   marginal carbon intensity windows (per the specified windowSize) for each, within the start and end time boundaries.
+    ///   If no start or end time boundaries are provided, all forecasted data points are used. 
+    ///
+    ///   The forecast data represents what the data source predicts future marginal carbon intesity values to be, 
+    ///   not actual measured emissions data (as future values cannot be known).
+    ///
+    ///   This endpoint is useful for determining if there is a more carbon-optimal time to use electicity predicted in the future.
+    /// </remarks>
+    /// <returns>An array of forecasts (one per requested location) with their optimal marginal carbon intensity windows.</returns>
+    /// <response code="200">Returns the requested forecast objects</response>
+    /// <response code="400">Returned if any of the input parameters are invalid</response>
+    /// <response code="500">Internal server error</response>
+    /// <response code="501">Returned if the underlying data source does not support forecasting</response>
     [Produces("application/json")]
     [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(IEnumerable<EmissionsForecastDTO>))]
     [ProducesResponseType(StatusCodes.Status400BadRequest, Type = typeof(ValidationProblemDetails))]
     [ProducesResponseType(StatusCodes.Status500InternalServerError, Type = typeof(ValidationProblemDetails))]
     [ProducesResponseType(StatusCodes.Status501NotImplemented, Type = typeof(ValidationProblemDetails))]
     [HttpGet("forecasts/current")]
-    public async Task<IActionResult> GetCurrentForecastData([FromQuery(Name = "location"), BindRequired] string[] locations, DateTimeOffset? startTime = null, DateTimeOffset? endTime = null, int? windowSize = null)
+    public async Task<IActionResult> GetCurrentForecastData([FromQuery(Name = "location"), BindRequired] string[] location, DateTimeOffset? startTime = null, DateTimeOffset? endTime = null, int? windowSize = null)
```

Comment:
the public API signature already uses the singular _location_, and IIRC the POR was for the 'internal code areas' of the 3+ APIs in this Controller that support to multiple locations, to keep using the plural _location**s**_

Reply:
The reason we changed this is because the swashbuckle swagger parser expects a "location" instead of "locations" in the query since it only cares about the internals so it was giving warnings so we thought it may be easier to just use location in this case.

Reply:
ok, 'fixed' this by reverting both the xml comment section and real param to _locations_. and swagger still "gets it right" with whats in _FromQuery_